### DISCUSSION TITLE
dcim.napalm_read_device issues

### DISCUSSION THREAD

**Discussion by resparis** [netbox-community/netbox#8709] [Q&A] [answered]
Hi, I am trying to use Napalm via Netbox but I am getting 
{
  "detail": "Error connecting to the device at 10.10.40.30: ConnectAuthError(10.10.40.30)"
}

How do i enable dcim.napalm_read_device to the user? 
Thanks

R

**Comment by candlerb** [comment, marked as answer]
That's not a Netbox permissions error: that's a Napalm error.  It means Napalm can't login to the device.

I suggest you get Napalm to work at the CLI first.  Once this is working, put the corresponding settings in `NAPALM_USERNAME`, `NAPALM_PASSWORD` and `NAPALM_ARGS` (if required) in `configuration.py`

**Reply by resparis** [reply]
Thanks! on top of that I had to create a permisison on the admin view for dcim.napalm_read_device.

**Reply by network-prometric** [reply]
how exactly did you get those  dcim.napalm_read_device permissions added ?

**Reply by candlerb** [reply]
https://docs.netbox.dev/en/stable/administration/permissions/

![image](https://user-images.githubusercontent.com/44789/180730359-31556ed8-0df5-4391-8cd5-378206f12db3.png)

Enter "napalm_read" in the Additional Actions box. (Note: not "napalm_read_device" nor "dcim.napalm_read_device". Those are internal values AFAIK)
